### ISSUE TITLE
Use rid3 pie example with cljs advanced code optimizations fails

### ISSUE THREAD

**Issue by orolle** [gadfly361/rid3#9]
Hi,
I try to run your pie example with cljs compiler with advanced optimizations turned on but the pie misses labels. When trying with figwheel in development mode, I do not have any problems.

It seems some functions are optimized away by cljs compiler when optimizations turned on. I tried to include cljsjs/d3 dependency and require cljsjs.d3 namespace but I could not get it to work with optimizations for the pie example. 

I believe the problem is caused by `(.centroid label d)` in text-label function.

**Comment by gadfly361** [comment]
@orolle I have found the externs on cljsjs for d3 are not comprehensive, which means (in some cases) you will need to add your own externs for advanced compilation. 

I have an extern for .centeroid here: https://github.com/gadfly361/rid3/blob/master/externs.js#L38

I used [lein-externs](https://github.com/ejlo/lein-externs) to generate the externs in that file.  I should note, that tool will create more externs than you actually need though.
